src/main/java/org/opensearch/security/dlic/rest/api/AuthFailureListenersApiAction.java

```diff
@@ -0,0 +1,288 @@
+                notFound(channel, "listener not found");
+            }
+            saveOrUpdateConfiguration(client, configuration, new OnSucessActionListener<>(channel) {
+                @Override
+                public void onResponse(IndexResponse indexResponse) {
+                    ok(channel, authFailureContent(config));
+                }
+            });
+        }).error((status, toXContent) -> response(channel, status, toXContent)))
+            .override(PUT, (channel, request, client) -> loadConfiguration(getConfigType(), false, false).valid(configuration -> {
+                ConfigV7 config = (ConfigV7) configuration.getCEntry(CType.CONFIG.toLCString());
+
+                String listenerName = request.param(NAME_JSON_PROPERTY);
+
+                ObjectNode body = (ObjectNode) DefaultObjectMapper.readTree(request.content().utf8ToString());
+                SecurityJsonNode authFailureListener = new SecurityJsonNode(body);
+                ValidationResult<SecurityJsonNode> validationResult = validateAuthFailureListener(authFailureListener, listenerName);
+
+                if (!validationResult.isValid()) {
+                    badRequest(channel, validationResult.toString());
+                    return;
+                }
+
+                // Try to put the listener by name
+                config.dynamic.auth_failure_listeners.getListeners()
+                    .put(listenerName, createAuthFailureListenerWithDefaults(authFailureListener));
+                saveOrUpdateConfiguration(client, configuration, new OnSucessActionListener<>(channel) {
+                    @Override
+                    public void onResponse(IndexResponse indexResponse) {
+
+                        ok(channel, authFailureContent(config));
+                    }
+                });
+            }).error((status, toXContent) -> response(channel, status, toXContent)));
+
+    }
+
+    private ConfigV7.AuthFailureListener createAuthFailureListenerWithDefaults(SecurityJsonNode authFailureListener) {
+        List<String> ignoreHosts = authFailureListener.get(IGNORE_HOSTS_JSON_PROPERTY).isNull()
+            ? Collections.emptyList()
+            : authFailureListener.get(IGNORE_HOSTS_JSON_PROPERTY).asList();
+
+        int allowedTries = authFailureListener.get(ALLOWED_TRIES_JSON_PROPERTY).isNull()
+            ? ALLOWED_TRIES_DEFAULT
+            : authFailureListener.get(ALLOWED_TRIES_JSON_PROPERTY).asInt();
+
+        int timeWindowSeconds = authFailureListener.get(TIME_WINDOW_SECONDS_JSON_PROPERTY).isNull()
+            ? TIME_WINDOW_SECONDS_DEFAULT
+            : authFailureListener.get(TIME_WINDOW_SECONDS_JSON_PROPERTY).asInt();
+
+        int blockExpirySeconds = authFailureListener.get(BLOCK_EXPIRY_JSON_PROPERTY).isNull()
+            ? BLOCK_EXPIRY_SECONDS_DEFAULT
+            : authFailureListener.get(BLOCK_EXPIRY_JSON_PROPERTY).asInt();
+
+        int maxBlockedClients = authFailureListener.get(MAX_BLOCKED_CLIENTS_JSON_PROPERTY).isNull()
```

Comment:
nit: is it possible to pass the default into the `asInt` call and move the null handling within?

i.e. reduce this call to `int maxBlockedClients = authFailureListener.get(MAX_BLOCKED_CLIENTS_JSON_PROPERTY).asInt(MAX_BLOCKED_CLIENTS_DEFAULT)`

Reply:
Good suggestion, it cleaned up the code a good amount. Did so in this commit: [d565320](https://github.com/opensearch-project/security/pull/4641/commits/d565320611db6e052685d837957b310256aa90db)